### ISSUE TITLE
[RFC] sort by weight

### ISSUE THREAD

**Issue by Geobert** [cobalt-org/cobalt.rs#671]
Initially, this was first talked about in https://github.com/cobalt-org/cobalt.rs/issues/395 so we can sort the docs by a "weight" field, in order to pin a post for example. 

I'm going to start the implementation of this, but before starting anything, I'm opening this RFC to be sure we are all on the same page on this :)

# My plan

I was going to introduce a new field `weight` (an `i32`) in the frontmatter.

After that, the work on sorting is already done in the pagination feature: https://github.com/cobalt-org/cobalt.rs/blob/ef83cfec90c9cef77f0ac508a7374ce4f4387a1b/src/pagination/mod.rs#L74

**Comment by epage** [comment]
Sounds good.  The only question I have is what the default weight should be.

**Comment by Geobert** [comment]
I was thinking to `0`, using `i32` we are in the middle of all the possible values
